### ISSUE TITLE
Enum Literal Handling Regression

### ISSUE THREAD

**Issue by max-muoto** [microsoft/pyright#7926]
To preface, apologies if there's a simpler reproducible example here, mostly trying to replicate what's happening across modules in my codebase. In Pyright 1.1.362, something like this would have been fine:

```python

import enum
from typing import TypeAlias

class MyEnum(enum.StrEnum):
    ARG_1 = "ARG_1"
    ARG_2 = "ARG_1"
    

MyEnumAlias: TypeAlias = MyEnum

def get_enum() -> MyEnumAlias: ...

z = get_enum()

def foo(enum: MyEnum): ...

foo(z.ARG_1)
```

That being, we alias an enum, and pass in value off the aliased type, now in 1.1.362, you'll get a warning when trying to pass in `ARG_1`:

In this example, it's obviously pointless to alias, in reality, we have an enum setting that's we're aliasing to change the typing of, think, `MY_ENUM as MyEnum`, for a wrapper we have on some project settings.

[Pyright Playground
](https://pyright-play.net/?pyrightVersion=1.1.363&pythonVersion=3.12&strict=true&code=FASwtgDg9gTgLgAgKYDsCuZgDMZTAuATwhBQHMFxp4EAVYpAQQBsQBDAZ2GAGNnOOCALKEAoujAAKVBgB0AZTgxxGAJQAuYAm0JGAJQDiAfQCMCALwIARPuMmrWnbaMAmC9ef3H27iJVgWdg51OgZAznc-CW4AEyQsBDIkOCMZKVUEAFoAPmExCXDghFkS7gAvdySUtMlVWPiELCgoaQkQqLUQktluJpay2U9VIA)

Version: 1.1.363

**Comment by erictraut** [comment]
I agree this is a bug, but I think the code sample is suspect and arguably should be considered a bug. Accessing enum members from an _instance_ of an enum happens to work with Python's enum class, but it's probably not intended.

Consider the following. Does this really make sense?

```python
print(MyEnum.ARG_1.ARG_2.ARG_1.ARG_2.ARG_1)
```

Perhaps your code sample intends to return `type[MyEnumAlias]` from `get_enum`?

I'll update pyright to match the runtime behavior even though I think it doesn't make much sense in this case.

**Comment by max-muoto** [comment]
> I agree this is a bug, but I think the code sample is suspect and arguably should be considered a bug. Accessing enum members from an _instance_ of an enum happens to work with Python's enum class, but it's probably not intended.
> 
> Consider the following. Does this really make sense?
> 
> ```python
> print(MyEnum.ARG_1.ARG_2.ARG_1.ARG_2.ARG_1)
> ```
> 
> Perhaps your code sample intends to return `type[MyEnumAlias]` from `get_enum`?
> 
> I'll update pyright to match the runtime behavior even though I think it doesn't make much sense in this case.

You're right! My code did mean to do that, unfortunately we're accessing some configuration settings through Django typed as `Any`, so didn't catch this. Thanks.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.364.
